spdx/relationship.py

```diff
@@ -58,53 +90,7 @@ def validate(self, messages):
     def validate_relationship(self, messages):
         messages = messages if messages is not None else []
         r_type = self.relationship.split(" ")[1]
-        if r_type not in [
-            None,
-            "AMENDS",
-            "OTHER",
-            "COPY_OF",
-            "TEST_OF",
-            "ANCESTOR_OF",
-            "BUILD_DEPENDENCY_OF",
-            "BUILD_TOOL_OF",
-            "CONTAINED_BY",
-            "CONTAINS",
-            "DATA_FILE_OF",
-            "DEPENDENCY_MANIFEST_OF",
-            "DEPENDENCY_OF",
-            "DEPENDS_ON",
-            "DESCENDANT_OF",
-            "DESCRIBED_BY",
-            "DESCRIBES",
-            "DEV_DEPENDENCY_OF",
-            "DEV_TOOL_OF",
-            "DISTRIBUTION_ARTIFACT",
-            "DOCUMENTATION_OF",
-            "DYNAMIC_LINK",
-            "EXAMPLE_OF",
-            "EXPANDED_FROM_ARCHIVE",
-            "FILE_ADDED",
-            "FILE_DELETED",
-            "FILE_MODIFIED",
-            "GENERATED_FROM",
-            "GENERATES",
-            "HAS_PREREQUISITE",
-            "METAFILE_OF",
-            "OPTIONAL_COMPONENT_OF",
-            "OPTIONAL_DEPENDENCY_OF",
-            "PACKAGE_OF",
-            "PATCH_APPLIED",
-            "PATCH_FOR",
-            "PREREQUISITE_FOR",
-            "PROVIDED_DEPENDENCY_OF",
-            "RUNTIME_DEPENDENCY_OF",
-            "STATIC_LINK",
-            "TEST_CASE_OF",
-            "TEST_DEPENDENCY_OF",
-            "TEST_TOOL_OF",
-            "VARIANT_OF",
-        ]:
-
+        if r_type not in [name for name, _ in RelationshipType.__members__.items()]:
```

Comment:
Instead of building a new list for storing all the enum-members, 
we could just have used the dictionary keys.

`RelationshipType.__members__` is a dictionary.

`if r_type not in RelationshipType.__members__.keys()` would be a preferred way for the following reasons:
1. `__members__` is a dictionary object, it's keys method returns a dictview object which takes amortized O(1) complexity to return a view object as opposed to O(N) for building a new list for all the members.
2. It is much more clean and shorter than the current method being used. The current method unnecessarily creates a new list object which consumes additional space and time.

Some might argue that we can even check membership using `RelationshipType._member_names_`. The reason I don't suggest using `_member_names` attribute is that I couldn't find that attribute in the latest official python documentation. Otherwise, it could've been more intuitive way to check membership.

Reply:
thanks for your advice Rishabh, could you please see this [issue](https://github.com/Yash-Varshney/tools-python/issues/2) which is the inspiration for this code.